worlds/witness/options.py

```diff
@@ -53,6 +63,61 @@ class ShuffleSymbols(DefaultOnToggle):
     display_name = "Shuffle Symbols"
 
 
+class ProgressiveSymbols(OptionSet):
+    """
+    Make some symbols progressive, if they exist.
+
+    By default, includes the chains where the second item can't be used without the first.
+
+    Progressive Dots: Dots -> Full Dots
+    Progressive Symmetry: Symmetry -> Colored Dots
+    Progressive Stars: Stars -> Stars + Same Colored Symbol
+    Progressive Squares: Black/White Squares -> Colored Squares
+    Progressive Shapers: Shapers -> Rotated Shapers -> Negative Shapers
+    Progressive Line-Counting Symbols: Triangles -> Arrows
+    """
+
+    valid_keys = {
+        "Progressive Dots",
+        "Progressive Symmetry",
+        "Progressive Stars",
+        "Progressive Squares",
+        "Progressive Shapers",
+        "Progressive Line-Counting Symbols"
+    }
+
+    default = {"Progressive Dots", "Progressive Symmetry", "Progressive Stars"}
+
+
+class SecondStageSymbolsActIndependently(Toggle):
+    """
+    Only relevant if Dots, Symmetry or Stars are not progressive.
+
+    If Dots are not progressive, this replaces "Dots" with "Sparse Dots", so that "Full Dots" can unlock Full Dots panels independently.
+    If Stars are not progressive, this replaces "Stars" with "Simple Stars", so that "Stars + Same Colored Symbol" can act independently.
+    Removes the Symmetry requirement from the Symmetry Laser panel sets so that Colored Dots can unlock something on their own.
+    """
+
+    visibility = Visibility.template | Visibility.complex_ui
+
+
+class ColoredDotsAreProgressiveDots(Toggle):

```

Comment:
Maybe I missed it, but if Dots aren't progressive this (and the other one) should throw an error/warning

Reply:
Hm, I could do a warning or an info, but there are things like "Obelisk Keys" which don't throw a warning when "EP Shuffle" is off, etc etc

Reply:
Hmm, that's true. I love how different worlds handle this totally differently, yeah maybe it's fine to just do nothing. Depends whether you think it's "incompatible to the point that the user likely is not expecting what they are about to get"